src/StateSmith/Output/Sim/HtmlRenderer.cs

```diff
@@ -4,7 +4,7 @@ namespace StateSmith.Output.Sim;
 
 public class HtmlRenderer
 {
-    public static void Render(StringBuilder stringBuilder, string smName, string mocksCode, string mermaidCode, string jsCode)
+    public static void Render(StringBuilder stringBuilder, string smName, string mocksCode, string mermaidCode, string jsCode, string diagramEventNamesArray)
```

Comment:
this makes sense. At some point we may need to do something similar for state names as well.
Right now, we're in control of the statenames because the mermaid is generated by us. But once we support using mermaid as an input language, that will no longer be true.
/tangent

Reply:
we can keep the same workflow when we support mermaid an input language if we want. Treat it just like we do for plantuml simulations right now. I haven't though far enough for a preference, just a thought.